diff --git chrome/browser/download/chrome_download_manager_delegate.cc chrome/browser/download/chrome_download_manager_delegate.cc
index 5ab37f0ec505..cf8f15483899 100644
--- chrome/browser/download/chrome_download_manager_delegate.cc
+++ chrome/browser/download/chrome_download_manager_delegate.cc
@@ -84,6 +84,8 @@
 #include "extensions/common/constants.h"
 #endif
 
+#include "cef/libcef/features/features.h"
+
 using content::BrowserThread;
 using download::DownloadItem;
 using content::DownloadManager;
@@ -267,6 +269,7 @@ ChromeDownloadManagerDelegate::ChromeDownloadManagerDelegate(Profile* profile)
           {base::MayBlock(), base::TaskPriority::BACKGROUND,
            base::TaskShutdownBehavior::CONTINUE_ON_SHUTDOWN})),
       weak_ptr_factory_(this) {
+#if !BUILDFLAG(ENABLE_CEF)
   DCHECK(!profile_->GetPath().empty());
   base::FilePath metadata_cache_file =
       profile_->GetPath().Append(chrome::kDownloadMetadataStoreFilename);
@@ -275,6 +278,7 @@ ChromeDownloadManagerDelegate::ChromeDownloadManagerDelegate(Profile* profile)
 #if defined(OS_ANDROID)
   location_dialog_bridge_.reset(new DownloadLocationDialogBridge);
 #endif
+#endif
 }
 
 ChromeDownloadManagerDelegate::~ChromeDownloadManagerDelegate() {
@@ -561,6 +565,7 @@ void ChromeDownloadManagerDelegate::ChooseSavePath(
 }
 
 download::InProgressCache* ChromeDownloadManagerDelegate::GetInProgressCache() {
+  return nullptr;
   DCHECK(download_metadata_cache_ != nullptr);
   return download_metadata_cache_.get();
 }
diff --git chrome/browser/ui/webui/constrained_web_dialog_ui.cc chrome/browser/ui/webui/constrained_web_dialog_ui.cc
index 0e061e83bf94..687e822a2550 100644
--- chrome/browser/ui/webui/constrained_web_dialog_ui.cc
+++ chrome/browser/ui/webui/constrained_web_dialog_ui.cc
@@ -27,6 +27,8 @@
 #include "chrome/browser/extensions/tab_helper.h"
 #endif
 
+#include "cef/libcef/features/features.h"
+
 using content::RenderFrameHost;
 using content::RenderViewHost;
 using content::WebContents;
@@ -57,7 +59,9 @@ class ConstrainedWebDialogDelegateUserData
 ConstrainedWebDialogUI::ConstrainedWebDialogUI(content::WebUI* web_ui)
     : WebUIController(web_ui) {
 #if BUILDFLAG(ENABLE_EXTENSIONS)
+#if !BUILDFLAG(ENABLE_CEF)
   extensions::TabHelper::CreateForWebContents(web_ui->GetWebContents());
+#endif 
 #endif
 }
 
diff --git chrome/browser/ui/webui/print_preview/print_preview_handler.cc chrome/browser/ui/webui/print_preview/print_preview_handler.cc
index 2913dbfdc781..68d59e122a17 100644
--- chrome/browser/ui/webui/print_preview/print_preview_handler.cc
+++ chrome/browser/ui/webui/print_preview/print_preview_handler.cc
@@ -75,6 +75,8 @@
 #include "printing/print_settings.h"
 #include "third_party/icu/source/i18n/unicode/ulocdata.h"
 
+#include "cef/libcef/features/features.h"
+
 #if defined(OS_CHROMEOS)
 #include "chrome/browser/chromeos/settings/device_oauth2_token_service.h"
 #include "chrome/browser/chromeos/settings/device_oauth2_token_service_factory.h"
@@ -467,14 +469,18 @@ void PrintPreviewHandler::RegisterMessages() {
 void PrintPreviewHandler::OnJavascriptAllowed() {
   // Now that the UI is initialized, any future account changes will require
   // a printer list refresh.
+#if !BUILDFLAG(ENABLE_CEF)
   RegisterForGaiaCookieChanges();
+#endif
 }
 
 void PrintPreviewHandler::OnJavascriptDisallowed() {
   // Normally the handler and print preview will be destroyed together, but
   // this is necessary for refresh or navigation from the chrome://print page.
   weak_factory_.InvalidateWeakPtrs();
+#if !BUILDFLAG(ENABLE_CEF)
   UnregisterForGaiaCookieChanges();
+#endif
 }
 
 WebContents* PrintPreviewHandler::preview_web_contents() const {
@@ -1194,6 +1200,9 @@ PrinterHandler* PrintPreviewHandler::GetPrinterHandler(
     }
     return privet_printer_handler_.get();
   }
+#else // !BUILDFLAG(ENABLE_SERVICE_DISCOVERY)
+  if (printer_type == PrinterType::kPrivetPrinter)
+    return nullptr;
 #endif
   if (printer_type == PrinterType::kPdfPrinter) {
     if (!pdf_printer_handler_) {
diff --git content/browser/download/download_manager_impl.cc content/browser/download/download_manager_impl.cc
index 52962a868999..506983be25df 100644
--- content/browser/download/download_manager_impl.cc
+++ content/browser/download/download_manager_impl.cc
@@ -344,7 +344,7 @@ DownloadManagerImpl::DownloadManagerImpl(BrowserContext* browser_context)
 }
 
 DownloadManagerImpl::~DownloadManagerImpl() {
-  DCHECK(!shutdown_needed_);
+  //DCHECK(!shutdown_needed_);
 }
 
 DownloadItemImpl* DownloadManagerImpl::CreateActiveItem(
diff --git ui/shell_dialogs/base_shell_dialog_win.cc ui/shell_dialogs/base_shell_dialog_win.cc
index 9ccd1bbc4562..23d14323fcb7 100644
--- ui/shell_dialogs/base_shell_dialog_win.cc
+++ ui/shell_dialogs/base_shell_dialog_win.cc
@@ -30,7 +30,7 @@ BaseShellDialogImpl::RunState BaseShellDialogImpl::BeginRun(HWND owner) {
   DCHECK(!IsRunningDialogForOwner(owner));
   // The owner must be a top level window, otherwise we could end up with two
   // entries in our map for the same top level window.
-  DCHECK(!owner || owner == GetAncestor(owner, GA_ROOT));
+  // DCHECK(!owner || owner == GetAncestor(owner, GA_ROOT));
   RunState run_state;
   run_state.dialog_thread = CreateDialogThread();
   run_state.owner = owner;
